packages/cli/src/commands/translate/utils.ts

```diff
@@ -53,69 +55,137 @@ export const createFullTranslation = async ({
+    }
+
+    void queue.add(async () => {
+      consoleLog.info(`Creating the translation for ${targetLocaleFile}`);
+      const result = await translate({
+        api: openai,
+        sourceFilePath: baseLocaleFile,
+        targetLanguage: languageTag,
+      });
 
-    const getTranslationPath = async () => {
-      if (existsSync(targetPath)) {
-        const currentContent = await fs.readFile(targetPath, 'utf8');
+      if (!result) {
+        consoleLog.fatal(`Unable to create the translation for ${targetLocaleFile}`);
+      }
+
+      await fs.mkdir(path.parse(targetLocaleFile).dir, { recursive: true });
+      await fs.writeFile(targetLocaleFile, result);
+      consoleLog.succeed(`The translation for ${targetLocaleFile} created`);
+    });
+  }
+
+  return queue.onIdle();
+};
+
+export const syncTranslation = async ({
+  instancePath,
+  packageName,
+  languageTag,
+  verbose = true,
+  queue = new PQueue({ concurrency: 5 }),
+}: TranslationOptions) => {
+  const localeFiles = await getBaseAndTargetLocaleFiles(instancePath, packageName, languageTag);
 
-        if (currentContent.includes('// UNTRANSLATED')) {
-          return targetPath;
-        }
+  if (verbose) {
+    consoleLog.info(
+      'Found ' +
+        String(localeFiles.length) +
+        ' file' +
+        conditionalString(localeFiles.length !== 1 && 's') +
+        ' in ' +
+        packageName +
+        ' to translate'
+    );
+  }
 
-        if (verbose) {
-          consoleLog.info(
-            `Target path ${targetPath} exists and has no untranslated mark, skipping`
-          );
-        }
+  const openai = createOpenaiApi();
 
-        return;
+  /* eslint-disable no-await-in-loop */
+  for (const { targetLocaleFile } of localeFiles) {
+    if (!existsSync(targetLocaleFile)) {
+      if (verbose) {
+        consoleLog.info(`Target locale file ${targetLocaleFile} does not exist, skipping`);
       }
 
-      return file;
-    };
+      continue;
+    }
 
-    const translationPath = await getTranslationPath();
+    const currentContent = await fs.readFile(targetLocaleFile, 'utf8');
 
-    if (!translationPath) {
+    if (!currentContent.includes(untranslatedMark)) {
+      if (verbose) {
+        consoleLog.info(
+          `Target path ${targetLocaleFile} exists and has no untranslated mark, skipping`
+        );
+      }
       continue;
     }
 
     void queue.add(async () => {
-      consoleLog.info(`Translating ${translationPath}`);
-      const result = await translate(openai, languageTag, translationPath);
+      consoleLog.info(`Translating ${targetLocaleFile}`);
+      const result = await translate({
+        api: openai,
+        sourceFilePath: targetLocaleFile,
+        targetLanguage: languageTag,
+        extraPrompt: `Object values without an "${untranslatedMark}" mark should be skipped and keep its original value. Remember to remove the "${untranslatedMark}" mark with the spaces before and after it in the output content.`,
```

Comment:
this will also increase a dozen of tokens.

Reply:
The max tokens in our prompt are about 1200; 2048 is enough.